compiler/src/iree/compiler/Utils/FlatbufferUtils.cpp

```diff
@@ -64,6 +66,20 @@ DenseIntElementsAttr FlatbufferBuilder::getBufferAttr(MLIRContext *context) {
       std::move(bufferData));
 }
 
+DenseIntElementsAttr FlatbufferBuilder::getHeaderPrefixedBufferAttr(
+    MLIRContext *context, uint32_t magic, uint32_t version) {
+  iree_flatbuffer_file_header_t header = {0};
+  header.magic = magic;
+  header.version = version;
+  header.content_size = flatcc_builder_get_buffer_size(*this);
+  auto bufferData = cloneBufferIntoContiguousBytes(
+      ArrayRef<uint8_t>((const uint8_t *)&header, sizeof(header)), *this);
```

Comment:
Can we do a `reinterpret_cast` instead? These are more greppable in case you suspects your casts to be off and want to audit all potentially unsafe ones